apache-rat-core/src/test/java/org/apache/rat/ReportTest.java

```diff
@@ -145,11 +176,193 @@ public void LicensesOptionTest() throws Exception {
+            lst.add(Arguments.of(args, test));
+        }
+
+//        Report LOG_LEVEL
+        DefaultLog log = (DefaultLog) DefaultLog.getInstance();
+        for (Log.Level l : Log.Level.values()) {
+            args = new String[]{"--" + Report.LOG_LEVEL.getLongOpt(), l.name().toLowerCase()};
+            test = c -> log.getLevel() == l;
+            lst.add(Arguments.of(args, test));
+        }
+//        Report.NO_DEFAULTS
+        args = new String[]{"--" + Report.NO_DEFAULTS.getLongOpt()};
+        test = c -> {
+            assertEquals(0, c.getLicenses(LicenseSetFactory.LicenseFilter.ALL).size());
+            return true;
+        };
+        lst.add(Arguments.of(args, test));
+
+//        Report.OUT;
+        File outFile = new File( tempDirectory, "outexample");
+        args = new String[] {"--"+Report.OUT.getLongOpt(), outFile.getAbsolutePath()};
+        test = c -> {
+            try (OutputStream os = c.getOutput().get()) {
+                os.write("Hello world".getBytes());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            try (BufferedReader reader = new BufferedReader( new InputStreamReader( new FileInputStream(outFile)))) {
+                return "Hello world".equals(reader.readLine());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        };
+        lst.add(Arguments.of(args, test));
+
+//        Report.SCAN_HIDDEN_DIRECTORIES
+        args = new String[]{"--" + Report.SCAN_HIDDEN_DIRECTORIES.getLongOpt()};
+        test = c -> c.getDirectoriesToIgnore() == FalseFileFilter.FALSE;
+        lst.add(Arguments.of(args, test));
+
+//        Report.STYLESHEET_CLI
+        test = c -> c.getStyleSheet() != null;
+        for (String sheet : new String[]{"plain-rat", "missing-headers", "unapproved-licenses"}) {
+            args = new String[]{"--" + Report.STYLESHEET_CLI.getLongOpt(), sheet};
+            lst.add(Arguments.of(args, test));
+        }
+        URL url = ReportTest.class.getResource("MatcherContainerResource.txt");
+        args = new String[]{"--" + Report.STYLESHEET_CLI.getLongOpt(), url.getFile()};
+        lst.add(Arguments.of(args, test));
+
+//        Report.XML;
+        args = new String[]{longOpt(Report.XML)};
+        test = ReportConfiguration::isStyleReport;
+        lst.add(Arguments.of(args, test.negate()));
+        return lst.stream();
+    }
+
+    @Test
+    public void testDeprecatedUseLogged() throws IOException {
+        TestingLog log = new TestingLog();
+        try {
+            DefaultLog.setInstance(log);
+            String[] args = {longOpt(Report.DIR), "foo", "-a"};
+            ReportConfiguration config = Report.parseCommands(args, (o) -> {
+            }, true);
+
+        } finally {
+            DefaultLog.setInstance(null);
+        }
+        log.assertContains("WARN: Option [-d, --dir] used.  Deprecated for removal since 0.17.0: Use '--'");
```

Comment:
Can you change this to 0.17 as we have 2digit version numbers only at the moment. This will make certain tests fail, but will be the correct version.